lib-ee/emqx_license/src/emqx_license_parser_v20220101.erl

```diff
@@ -103,35 +109,49 @@ parse_payload(Payload) ->
     end.
 
 parse_type(TypeStr) ->
-    case string:to_integer(TypeStr) of
-        {Type, <<"">>} -> {ok, Type};
+    case parse_int(TypeStr) of
+        {ok, Type} -> {ok, Type};
         _ -> {error, invalid_license_type}
     end.
 
 parse_customer_type(CTypeStr) ->
-    case string:to_integer(CTypeStr) of
-        {CType, <<"">>} -> {ok, CType};
+    case parse_int(CTypeStr) of
+        {ok, CType} -> {ok, CType};
         _ -> {error, invalid_customer_type}
     end.
 
-parse_date_start(<<Y:4/binary, M:2/binary, D:2/binary>>) ->
-    Date = list_to_tuple([N || {N, <<>>} <- [string:to_integer(S) || S <- [Y, M, D]]]),
-    case calendar:valid_date(Date) of
-        true -> {ok, Date};
-        false -> {error, invalid_date}
+parse_date_start(DateStr) ->
+    case parse_int(DateStr) of
+        {ok, Num} when Num >= ?MIN_START_DATE ->
+            Y = Num div 1_00_00,
+            M = (Num rem 1_00_00) div 1_00,
+            D = Num rem 1_00,
+            case calendar:valid_date({Y, M, D}) of
+                true -> {ok, {Y, M, D}};
+                false -> {error, invalid_date}
+            end;
+        _ ->
+            {error, invalid_date}
     end;
 parse_date_start(_) -> {error, invalid_date}.
 
 parse_days(DaysStr) ->
-    case string:to_integer(DaysStr) of
-        {Days, <<"">>} when Days > 0 -> {ok, Days};
+    case parse_int(DaysStr) of
+        {ok, Days} when Days > 0 -> {ok, Days};
         _ -> {error, invalid_int_value}
     end.
 
 parse_max_connections(MaxConnStr) ->
-    case string:to_integer(MaxConnStr) of
-        {MaxConns, <<"">>} when MaxConns > 0 -> {ok, MaxConns};
-        _ -> {error, invalid_int_value}
+    case parse_int(MaxConnStr) of
+        {ok, MaxConns} when MaxConns > 0 -> {ok, MaxConns};
+        _ -> {error, invalid_connection_limit}
+    end.
+
+parse_int(Str0) ->
+    Str = iolist_to_binary(string:replace(Str0, ",", "")),
```

Comment:
Just an observation: a few locales use `.` as a thousands delimiter as well, maybe that could be removed too?

Reply:
maybe keep it so not to confuse with float point numbers 

allow `,` is to make it easier for license managers. not EMQX users.